Code/Framework/AzCore/AzCore/Settings/ConfigParser.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/IO/GenericStreams.h>
+#include <AzCore/Settings/ConfigParser.h>
+#include <AzCore/std/string/conversions.h>
+#include <AzCore/std/string/fixed_string.h>
+#include <AzCore/std/containers/fixed_vector.h>
+#include <AzCore/StringFunc/StringFunc.h>
+
+namespace AZ::Settings
+{
+    AZStd::string_view ConfigParserSettings::DefaultCommentPrefixFilter(AZStd::string_view line)
+    {
+        constexpr AZStd::string_view commentPrefixes = ";#";
+        for (char commentPrefix : commentPrefixes)
+        {
+            if (size_t commentOffset = line.find(commentPrefix); commentOffset != AZStd::string_view::npos)
+            {
+                line = line.substr(0, commentOffset);
+            }
+        }
+
+        return line;
+    }
+
+    AZStd::string_view ConfigParserSettings::DefaultSectionHeaderFilter(AZStd::string_view line)
+    {
+        AZStd::string_view sectionName;
+        constexpr char sectionHeaderStart = '[';
+        constexpr char sectionHeaderEnd = ']';
+        if (line.starts_with(sectionHeaderStart) && line.ends_with(sectionHeaderEnd))
```

Comment:
When I reached this code I wondered what happens if there are trailing spaces, then I saw in the parsing loop that for each line the trailing spaces are removed before calling this function. Maybe it'd be good to add a small clarification note here indicating that the line received by argument won't have trailing spaces.

Reply:
Well that is the normal part of line parsing that leading and trailing whitespace are not signficant or kept.
I'll add a comment noting as such